compass/machines/cori-haswell.cfg

```diff
@@ -20,6 +20,9 @@ compass_envs = /global/common/software/e3sm/compass/cori-haswell/base
 # machines
 [deploy]
 
+# the name of the machine
+machine = cori-haswell
```

Comment:
@xylar , based on our discussion last week, I made changes to this PR to use the standard compass job generation function instead of having a special copy in this test case.  I was able to do that with only one change to compass framework, which is this line letting the config know the machine name being used.  This is because the `write_job_script` function requires `machine` as an input argument, and I could not see anywhere that the machine is saved and made accessible other than through the `compass setup` call stack (which was not accessible from my step's `setup` method).  

I could imagine there are other ways this problem could be resolved, so I want to get your feedback on my adding this line to the machine .cfg file before I add the corresponding lines to the other machine cfg files.  You certainly would not want users to modify this cfg entry, but there are a number of other entries that have the same situation (e.g. compiler).  On the other hand, I could imagine other uses for saving the machine name (e.g. a test case wanting to do things differently depending on which machine it is being run on).

Commit f894a044e4e012e73c98f444c5095c4d3bdf4dea explains why this was needed and shows how it was used to solve my problem.

Reply:
@matthewhoffman, I think it might be better to just add the machine name in code during setup.  That way, it doesn't have to be included by hand in each machine's config file, especially because it's redundant.

I would add the following here in the code:
https://github.com/matthewhoffman/compass/blob/7b02c0c924ee77babe975afea4bf48e846360733/compass/setup.py#L434-L437
```python
    # add the compass machine config file
    if machine is None:
        machine = 'default'
    config.add_from_package('compass.machines', f'{machine}.cfg')

    if machine is not None:
        config.set('deploy', 'machine', machine)
```

Reply:
@xylar , yes I agree that sound cleaner.  I'll add that.  Thanks for the feedback.

Reply:
Changed in c1069fe9b14fcc5fc4d892c08687a0eaa8d39bf7